qt/Utils.cc

```diff
@@ -301,3 +302,31 @@ QColor Utils::getFadedColor(QColor const& color)
     faded_color.setAlpha(128);
     return faded_color;
 }
+
+QString Utils::getDynamicDownloadDir(Prefs const& prefs, QString const& torrentName)
```

Comment:
I'd _strongly_ prefer that we match not just on name but also announceUrl, tracker, e.g. "if this torrent is from Tracker A, put it in /path/to/trackers/A" because that is a use case that I would use constantly :slightly_smiling_face: 

This would take some extra work in AddData to get a URL accessor from parsed metainfo / peeled out of the magnet URL etc.